pkg/cloud/cloud.go

```diff
@@ -1621,25 +1677,53 @@ func (c *cloud) waitForVolumeModification(ctx context.Context, volumeID string)
 	return nil
 }
 
+func describeVolumesModifications(ctx context.Context, svc EC2API, request *ec2.DescribeVolumesModificationsInput) ([]types.VolumeModification, error) {
+	volumeModifications := []types.VolumeModification{}
+	var nextToken *string
+	for {
+		response, err := svc.DescribeVolumesModifications(ctx, request)
+		if err != nil {
+			if isAWSErrorModificationNotFound(err) {
+				return nil, VolumeNotBeingModified
+			}
+			return nil, fmt.Errorf("error describing volume modifications: %w", err)
+		}
+
+		volumeModifications = append(volumeModifications, response.VolumesModifications...)
+
+		nextToken = response.NextToken
+		if aws.ToString(nextToken) == "" {
+			break
+		}
+		request.NextToken = nextToken
+	}
+	return volumeModifications, nil
+}
+
 // getLatestVolumeModification returns the last modification of the volume.
-func (c *cloud) getLatestVolumeModification(ctx context.Context, volumeID string) (*types.VolumeModification, error) {
+func (c *cloud) getLatestVolumeModification(ctx context.Context, volumeID string, isBatched bool) (*types.VolumeModification, error) {
```

Comment:
np: rename `isBatched` to something like `batchable`, because it's possible for `isBatched` to be true but for the volume to not be batched (if batching is globally disabled).